test/services/hyperlink-utility.test.js

```diff
@@ -0,0 +1,164 @@
+    this.args = [{ name: 'arg_1', description: 'argument description' }];
+  }
+}
+const testThisConfig = ({ platform, env, argv, stream }) => {
+  platform = platform || 'darwin';
+  env = env || {};
+  argv = argv || [];
+  // back up the original env
+  const oldPlatform = process.platform;
+  const oldEnv = process.env;
+  const oldArgv = process.argv;
+
+  // Inject new env properties from args
+  Object.defineProperties(process, {
+    platform: { value: platform },
+    env: { value: env },
+    argv: { value: [process.argv[0], ...argv] },
+  });
+
+  const result = supportsHyperlink(stream);
+  // restore the original env
+  Object.defineProperties(process, {
+    platform: { value: oldPlatform },
+    env: { value: oldEnv },
+    argv: { value: oldArgv },
+  });
+  return result;
+};
+
+const testLink = test
+  .loadConfig()
+  .add('help', (ctx) => new TwilioHelp(ctx.config))
+  .register('cmdTestLink', (args) => {
+    return {
+      async run(ctx) {
+        const dummyHelpCommand = new TestCommand(args, ctx.config);
+        dummyHelpCommand.docLink = 'https://twilio.com/docs/dummyCmd';
+        const help = ctx.help.formatCommand(dummyHelpCommand);
+        ctx.cmdTestLink = help
+          .split('\n')
+          .map((s) => s.trimRight())
+          .join('\n');
+      },
+    };
+  });
+
+describe('supportsHyperlink', () => {
+  describe('test hyperlink generation', () => {
+    describe('test for Mac terminals', () => {
+      afterEach(() => {
+        flush();
+      });
+      test.it('not supported in Mac Terminal', () => {
+        expect(
+          testThisConfig({
+            env: {
+              TERM_PROGRAM: '',
+            },
+            stream: {
+              isTTY: false,
+            },
+          }),
+        ).to.be.false;
+      });
+      test.it('testing convertToHyperlink, supported iTerm.app 3.1, tty stream', () => {
+        const result = testThisConfig({
+          env: {
+            TERM_PROGRAM: 'iTerm.app',
+            TERM_PROGRAM_VERSION: '3.1.0',
+          },
+          stream: {
+            isTTY: true,
+          },
+        });
+        if ('CI' in process.env) {
+          expect(result).to.be.false;
+          expect(convertToHyperlink('MORE INFO', 'https://twilio.com/docs/dummyCmd').isSupported).to.be.false;
+        } else {
+          expect(result).to.be.true;
+          expect(convertToHyperlink('MORE INFO', 'https://twilio.com/docs/dummyCmd').isSupported).to.be.true;
+        }
+      });
+    });
+
+    describe('test for iTerm terminals', () => {
+      afterEach(() => {
+        flush();
+      });
+      test.it('supported in iTerm.app 3.1, tty stream', () => {
+        const result = testThisConfig({
+          env: {
+            TERM_PROGRAM: 'iTerm.app',
+            TERM_PROGRAM_VERSION: '3.1.0',
+          },
+          stream: {
+            isTTY: true,
+          },
+        });
+        if ('CI' in process.env) {
```

Comment:
The supports-hyperlink module [discourages](https://github.com/jamestalmage/supports-hyperlinks/blob/66834a8a4ccc07f9af6364778fcaa0666b61eccb/index.js#L56) CI, because of which the test-cases would return false in a CI environment but would work fine in a local setup. Therefore, the if-else.

Reply:
Isn't  it possible to remove and reassign `CI` env before/after each test to make this also testable in a CI environment?

Reply:
Yeah I tried that and realised that the CI env variable and other added env variables (for [stream](https://github.com/jamestalmage/supports-hyperlinks/blob/66834a8a4ccc07f9af6364778fcaa0666b61eccb/index.js#L40) and [Ubuntu](https://github.com/jamestalmage/supports-hyperlinks/blob/66834a8a4ccc07f9af6364778fcaa0666b61eccb/index.js#L78) possibly) caused the tests to fail, and resetting them all didn't feel like a viable option then. I am up for giving it a try if you feel that could be better.

Reply:
The link to [commit](https://github.com/twilio/twilio-cli/commit/8a81da568306422bc599d882699fbe46d85e938e) and [build](https://app.travis-ci.com/github/twilio/twilio-cli/builds/237514212) that failed with reassigning CI env variable.